apps/autoscheduler/CMakeLists.txt

```diff
@@ -36,7 +36,7 @@ set_target_properties(retrain_cost_model_process
 target_include_directories(retrain_cost_model_process
                            PRIVATE ${CMAKE_CURRENT_SOURCE_DIR}/../support)
 target_link_libraries(retrain_cost_model_process
-                      PRIVATE cost_model train_cost_model)
+                      PRIVATE cost_model train_cost_model Halide)
```

Comment:
This should not need to link in libHalide -- in general, you should only need libHalide if you are using the JIT (or you are a Generator). Why did you need to add this?

Reply:
Without the libHalide dependency, symbols corresponding to multiple methods of the Halide::Internal::Introspection and Halide::Internal::ErrorReport::ErrorReport classes are missing. I picked up the dependency through the #include "Error.h" in FunctionDAG.h. I'm guessing that the introspection dependency came through the added dependency on the Error class.

Reply:
Ouch. We need to not do that. It is a bad code smell to need to link in libHalide just for error reporting.